src/index.ts

```diff
@@ -1,19 +1,649 @@
+    transaction: Transaction,
+    // TODO: Why is this required? Could we instantiate this ourselves with the gateway param just like ArConnect?
+    arweave: Arweave,
+    node?: string,
+  ): Promise<DispatchResult> {
+    const publicKeyBuffer = this.auth0.getCachedUserPublicKeyBuffer();
+    const sub = this.auth0.getCachedUserSub();
+
+    if (!publicKeyBuffer || !sub) throw new Error("Missing cached user.");
+
+    const signer: Signer = {
+      publicKey: publicKeyBuffer,
+      signatureType: 1,
+      signatureLength: 512,
+      ownerLength: 512,
+      sign: this.api.getSignerSignFn(sub),
+      // Note we don't provide `verify` as it's not used anyway:
+      // verify: null,
+    };
+
+    addOthentAnalyticsTags(transaction);
+
+    // Using transaction.tags won't work as those wound still be encoded:
+    const tags = (transaction.get("tags") as unknown as Tag[]).map((tag) => ({
+      name: tag.get("name", { decode: true, string: true }),
+      value: tag.get("value", { decode: true, string: true }),
+    }));
+
+    const dateItem = createData(transaction.data, signer, { tags });
+
+    // DataItem.sign() sets the DataItem's `id` property and returns its `rawId`:
+    await dateItem.sign(signer);
+
+    try {
+      // TODO: Try with a bunch of different nodes?
+      if (!node) {
+        node = "https://turbo.ardrive.io";
+      }
+
+      const res = await fetch(`${node}/tx`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/octet-stream",
+        },
+        body: Buffer.from(dateItem.getRaw()),
+      });
+
+      if (res.status >= 400) {
+        throw new Error(
+          `Error uploading DataItem: ${res.status} ${res.statusText}`,
+        );
+      }
+
+      return {
+        // TODO: Should we return the dataItem itself as well?
+        id: await dateItem.id,
+      };
+    } catch {
+      await this.sign(transaction);
+
+      const uploader = await arweave.transactions.getUploader(transaction);
+
+      while (!uploader.isComplete) {
+        await uploader.uploadChunk();
+      }
+
+      return {
+        id: transaction.id,
+      };
+    }
+  }
+
+  // ENCRYPT/DECRYPT:
+
+  /**
+   * Encrypt data with the users JWK. This function assumes (and requires) a user is logged in and a valid string to sign.
+   * @param plaintext The data in string format to sign.
+   * @returns The encrypted data.
+   */
+  async encrypt(plaintext: string | BinaryDataType): Promise<Uint8Array> {
+    const sub = this.auth0.getCachedUserSub();
+
+    if (!sub) throw new Error("Missing cached user.");
+
+    const ciphertextBuffer = await this.api.encrypt(plaintext, sub);
+
+    return ciphertextBuffer;
+  }
+
+  /**
+   * Decrypt data with the users JWK. This function assumes (and requires) a user is logged in and a valid encrypt() response.
+   * @param ciphertext The data to decrypt.
+   * @returns The decrypted data.
+   */
+  async decrypt(
+    ciphertext: string | BinaryDataType,
```

Comment:
In the old version of the SDK, `encrypt` would return a `BufferObject`, which could be passed directly into `decrypt`. For that to work now (e.g. if someone decided to just store those as-is), we would need to add this line:

    const parsedCiphertext = isBufferObject(ciphertext) ? new Uint8Array(ciphertext.data) : ciphertext;